opennms-services/src/main/java/org/opennms/protocols/jmx/connectors/Jsr160ConnectionFactory.java

```diff
@@ -222,11 +223,23 @@ private static JMXServiceURL getUrl(InetAddress address, int port, String protoc
             // Create an JMXMP connector client and
             // connect it to the JMXMP connector server
             //
-            url = new JMXServiceURL(protocol, InetAddressUtils.str(address), port, urlPath);
+            url = new JMXServiceURL(protocol, toUrlIpAddress(address), port, urlPath);
         } else {
             // Fallback, building a URL for RMI
-            url = new JMXServiceURL("service:jmx:" + protocol + ":///jndi/" + protocol + "://" + InetAddressUtils.str(address) + ":" + port + urlPath);
+            url = new JMXServiceURL("service:jmx:" + protocol + ":///jndi/" + protocol + "://" + toUrlIpAddress(address) + ":" + port + urlPath);
         }
         return url;
-    }    
+    }
+
+    /**
+     * Method that wraps IPv6 addresses in square brackets so that they are parsed
+     * correctly by the {@link JMXServiceURL} class.
+     */
+    private static String toUrlIpAddress(InetAddress addr) {
```

Comment:
I'm going to have to move this method to ```InetAddressUtils``` since the provisioning detectors also need it but it will still be a separate ```toUrlIpAddress()``` method.